### DISCUSSION TITLE
FlatButton

### DISCUSSION THREAD

**Discussion by Drioppi** [balta-io/forum#72] [Tira Dúvidas] [answered]
Bom dia!
No curso de Layouts ao incluir o FlatButton apresenta erro e pra corrigir tenho que criar uma função.

![image](https://user-images.githubusercontent.com/116727435/199014850-ba88b312-29e2-44fc-8614-5948e59519ef.png)

**Comment by JeanMRocha** [comment]
O FlatButton está deprecated.
Minha solução foi passar o FlatButton para TextButton que é o novo padrão de botão: meu código ficou assim. 

Vai dar erro também na key (o mais difícl de resolver que eu achei) que no caso não passa automaticamente, cria o tópico que eu posto a solução também. Eu demorei 4 dias estudando para achar os novos padrões, comecei no flutter tem uns 30 dias e estou apanhando demais.

            child: TextButton(
              style: TextButton.styleFrom(
                shape: const BeveledRectangleBorder(
                    borderRadius: BorderRadius.all(Radius.circular(15))),
                backgroundColor: Colors.blue,
              ),
              onPressed: func,
              child: Text(
                text,
                style: TextStyle(
                    color:
                        invert ? Colors.white : Theme.of(context).primaryColor,
                    fontSize: 25,
                    fontFamily: "Big Shoulders Display"),
              ),
            ),

**Comment by JeanMRocha** [comment, marked as answer]
> Bom dia! No curso de Layouts ao incluir o FlatButton apresenta erro e pra corrigir tenho que criar uma função.
> 
> ![image](https://user-images.githubusercontent.com/116727435/199014850-ba88b312-29e2-44fc-8614-5948e59519ef.png)

Abre lá no fórum do projeto as dicussões pois acho que este fórum aqui é geral.

link do forum 
https://github.com/balta-io/7192/discussions

**Reply by andrebaltieri** [reply]
Bom dia, @Drioppi agora é apenas Button... Mudou nas novas versões do Flutter.
https://docs.flutter.dev/release/breaking-changes/buttons

Seria bacana utilizar a mesma versão do curso, depois você tenta fazer nas novas versões...
